sway-ir/src/value.rs

```diff
@@ -23,6 +23,7 @@ pub struct Value(pub generational_arena::Index);
 pub struct ValueContent {
     pub value: ValueDatum,
     pub span_md_idx: Option<MetadataIndex>,
+    pub state_idx_md_idx: Option<MetadataIndex>,
```

Comment:
These two should really be replaced with a `Vec` but I'm planning to do that in a separate PR because a lot of code would have to change which would be distracting.